services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/docker_utils.py

```diff
@@ -236,3 +239,23 @@ async def _pull_image_with_progress(
             await progress_cb(total_current, total_total)
 
         await log_cb(f"pulling {shorter_image_name}: {pull_progress}...")
+
+
+@cached()
+async def supports_volumes_with_quota() -> bool:
+    async with docker_client() as docker:
+        docker_volume: Optional[DockerVolume] = None
+        try:
+            docker_volume = await docker.volumes.create(
+                {
+                    "name": f"check-quota-{uuid4()}",
+                    "Driver": "local",
+                    "DriverOpts": {"size": "1m"},
+                }
+            )
+        except aiodocker.DockerError:
+            logger.debug("No support for volume with quota")
+            return False
+        if docker_volume:
+            await docker_volume.delete()
```

Comment:
what if delete raises?

Reply:
THOUGHT: you want to check whether your volume has quotas and for that **you perform a test**. If the test fails in a very specific way, you conclude it has not quotas. OK.
Now if the test fails for other reason or during cleanup (i.e. docker_volume.delete()), it raises an exception that probably ends up as a 500 that reaches the client.  The client should interpret that as "cannot know whether it has quotas or not".

If it raises **during cleanup**  (i.e. ``docker_volume.delete()``) I find that the client got the the wrong interpretation since the actual test you were intended to do succeeded in the first place. What went wrong is something else that has nothing to do with the answer that the client is interested in.

IMO that exception should be handled in place. Moreoever, i would do it in a background task as a ``on_cleanup`` event independent from the request.

Reply:
Let's see how this will behave, moved it to a background task.